File: Normal/0232/main.py
from typing import List


class MyQueue:
    stack: List[int]
    def __init__(self):
        self.stack = []

    def push(self, x: int) -> None:
        self.stack.append(x)

    def pop(self) -> int:
        length = len(self.stack)
        if length <= 0:
            return 0
        firstValue = self.stack[0]
        self.stack = self.stack[1: length]
        return firstValue

    def peek(self) -> int:
        if len(self.stack) == 0:
            return 0
        return self.stack[0]

    def empty(self) -> bool:
        return len(self.stack) == 0



# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: Normal/0232/main.go
package main

type MyQueue struct {
	Stack []int
}

func Constructor() MyQueue {
	return MyQueue{[]int{}}
}

func (this *MyQueue) Push(x int) {
	this.Stack = append(this.Stack, x)
}

func (this *MyQueue) Pop() int {
	length := len(this.Stack)
	if length <= 0 {
		return 0
	}
	firstValue := this.Stack[0]
	this.Stack = this.Stack[1:length]
	return firstValue
}

func (this *MyQueue) Peek() int {
	if len(this.Stack) <= 0 {
		return 0
	}
	return this.Stack[0]
}

func (this *MyQueue) Empty() bool {
	return len(this.Stack) == 0
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Peek();
 * param_4 := obj.Empty();
 */
